lib/tls/aws_tls.c

```diff
@@ -675,7 +771,37 @@ BaseType_t TLS_Connect( void * pvContext )
          * runtime configuration should use verbose output. */
         mbedtls_ssl_conf_dbg( &pxCtx->xMbedSslConfig, prvTlsDebugPrint, NULL );
         mbedtls_debug_set_threshold( tlsDEBUG_VERBOSE );
-    #endif
+
+        /* Wireshark debugging support on the Windows Simulator.
+         * When https://github.com/ARMmbed/mbedtls/issues/1504 is implemented consider
+         * deleting this code. To give proper credit, this code is adapted from:
+         * https://github.com/Lekensteyn/mbedtls/commit/68aea15. */
+        #ifdef WIN32
+            {
+                /* On windows an environment variable value has a maximum size
+                 * of 32,627 characters, including the null-terminating character. */
+                char cKeyLogFileName[ 32767 ];
+                unsigned int ulReturnValue = GetEnvironmentVariable(
```

Comment:
I don't think it's safe to make Win32 API calls from a FreeRTOS task (i.e. in WinSim). The problem is that the Win32 thread scheduler and the FreeRTOS task scheduler don't know about each other. For example, it is reasonably common for an API call such as this to (a) take the loader lock and (b) block the thread, at least the first time it's called. Since the FreeRTOS scheduler doesn't know that this app task just got blocked, it no longer has coherent state. I also suspect that this could deadlock. 

I think our only option for supporting this feature is via the logging task.

Reply:
Or perhaps I can grab the environment variable in main.c? Or I can add it as a TLS config parameter?